zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/command/ActivateJobsCommandImpl.java

```diff
@@ -25,78 +25,111 @@
 import io.camunda.zeebe.client.api.command.FinalCommandStep;
 import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
 import io.camunda.zeebe.client.impl.RetriableStreamingFutureImpl;
+import io.camunda.zeebe.client.impl.http.HttpClient;
+import io.camunda.zeebe.client.impl.http.HttpZeebeFuture;
 import io.camunda.zeebe.client.impl.response.ActivateJobsResponseImpl;
+import io.camunda.zeebe.client.protocol.rest.JobActivationRequest;
+import io.camunda.zeebe.client.protocol.rest.JobActivationResponse;
 import io.camunda.zeebe.gateway.protocol.GatewayGrpc.GatewayStub;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsRequest;
 import io.camunda.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsRequest.Builder;
 import io.grpc.stub.StreamObserver;
 import java.time.Duration;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
+import org.apache.hc.client5.http.config.RequestConfig;
 
 public final class ActivateJobsCommandImpl
     implements ActivateJobsCommandStep1, ActivateJobsCommandStep2, ActivateJobsCommandStep3 {
 
   private static final Duration DEADLINE_OFFSET = Duration.ofSeconds(10);
   private final GatewayStub asyncStub;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
   private final JsonMapper jsonMapper;
   private final Predicate<StatusCode> retryPredicate;
-  private final Builder builder;
+  private final Builder grpcRequestObjectBuilder;
+  private final JobActivationRequest httpRequestObject;
   private Duration requestTimeout;
+  private boolean useRest;
 
   private final Set<String> defaultTenantIds;
   private final Set<String> customTenantIds;
 
   public ActivateJobsCommandImpl(
       final GatewayStub asyncStub,
+      final HttpClient httpClient,
       final ZeebeClientConfiguration config,
       final JsonMapper jsonMapper,
-      final Predicate<StatusCode> retryPredicate) {
+      final Predicate<StatusCode> retryPredicate,
+      final boolean useRest) {
```

Comment:
🙃 Do we need the `useRest` flag in the constructor when we already pass the whole `ZeebeClientConfiguration` object which contains it? It's more readable this way though.